### PR TITLE
Allow option to attach managed policies to roles

### LINKED ISSUE #14: Allow option to attach managed policies to all the roles
**Is this a BUG REPORT or FEATURE REQUEST?**:
FEATURE REQUEST

**What happened**:
There might be a use case where organizations wants to add one or more policies to all the roles and we should allow that to be attached to all the roles managed by iam-manager.

**What you expected to happen**:
iam-manager must attach all the managed policies provided to all the roles with out user providing it explicitly for every role

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- iam-manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:

```
- controller logs:
```
$ kubectl logs <iam-manager-pod>
```

### PR DESCRIPTION
**Pull request by kshamajain99**
close #14 

**Could you share the solution in high level?**

- [x] Attach administrator provided managed policies while creating role
- [x] While deleting role detach administrator provided managed policies

**Could you share the test results?**

Screen Shot 2020-01-28 at 1 03 55 PM

### CODE REVIEW

File: internal/config/properties.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ const (
 	iamPolicyS3Restricted = "iam.policy.s3.restricted.resource"
 	awsAccountId          = "aws.accountId"
 	awsMasterRole         = "aws.MasterRole"
+	managedPolicies       = "iam.managed.policies"
```
Comment (mnkg561) [inline internal/config/properties.go]: config map property name and this field is not matching

File: pkg/awsapi/iam.go [hunk 2]
```diff
@@ -104,12 +109,18 @@ func (i *IAM) CreateRole(ctx context.Context, req IAMRoleRequest) (*IAMRoleRespo
...
+	//Attach inline role policy
 	err = i.AddPermissionBoundary(ctx, req)
 
 	if err != nil {
 		return &IAMRoleResponse{}, err
 	}
 
+	//Attach managed role policy
+	for _, policy := range ManagedPolicies {
+		err = i.AttachManagedRolePolicy(ctx, policy, req.Name)
```
Comment (mnkg561) [inline pkg/awsapi/iam.go]: You should return the error and log the error

File: pkg/awsapi/iam.go [hunk 3]
```diff
@@ -292,9 +302,47 @@ func (i *IAM) AttachInlineRolePolicy(ctx context.Context, req IAMRoleRequest) (*
...
+		return err
+	}
+	return nil
+}
+
 //DeleteRole function deletes the role in the account
 func (i *IAM) DeleteRole(ctx context.Context, roleName string) error {
 
+	// Detach managed policies
+	for _, policy := range ManagedPolicies {
```
Comment (mnkg561) [inline pkg/awsapi/iam.go]: Check out a way if you can get the list of managed policies attached to the role? There might be a cases where people updating the role in the AWS IAM console directly and that leaves us in error state if we try to delete the role.
Check out https://github.com/keikoproj/iam-manager/issues/11. we can probably fix this issue as part of that

File: api/v1alpha1/zz_generated.deepcopy.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// autogenerated by controller-gen object, do not modify manually
+// Code generated by controller-gen. DO NOT EDIT.
```
Comment (mnkg561) [inline api/v1alpha1/zz_generated.deepcopy.go]: What is the go lang version in your machine? Lets see why it is changed that comment in your environment

File: internal/config/properties.go (hunk 1 above)
Comment (kshamajain99) [reply internal/config/properties.go]: It was supposed to be "iam.managed.policies", I will change it in config map.

File: api/v1alpha1/zz_generated.deepcopy.go (hunk 4 above)
Comment (kshamajain99) [reply api/v1alpha1/zz_generated.deepcopy.go]: go version go1.13.1 darwin/amd64

File: pkg/awsapi/iam.go (hunk 3 above)
Comment (kshamajain99) [reply pkg/awsapi/iam.go]: I will do that

**Pull request merged**